### ISSUE TITLE
Disallow role button on summary element

### ISSUE THREAD

**Issue by lemnis** [w3c/html-aria#85]
I do not know if this is discussed previously.

Currently spec states at the `summary` element:

>   Role: button with aria-expanded="true" if the parent (details) element's open attribute is present, aria-expanded="false" otherwise.

When I am reading this particular sentence I conclude that this is allowed:
```html
<details open>
  <summary role='button' aria-expanded='true'>A title</summary>
  ...
</details>
```

I have two reasons why the sentence it should be removed,
* <s>at `input[type='checkbox']` it also doesn't state that `aria-checked` should reflect the current `checked` state, so I would expect that the `aria-expanded` also should not be explained at this place.</s>
* implicit roles may not be used as stated earlier in the document, but the sentence make it look like it is allowed at the summary element

---

It would prefer a note as used at `input[type='checkbox']`, e.g.  

**Note**: if the parent (details) element's open attribute is present it should have `aria-expanded="true"`, `aria-expanded="false"` otherwise

**Comment by stevefaulkner** [comment]
The reason why this is currently allowed is that not all browsers support (edge and IE) details/summary, so in order to provide the appropriate semantics for JavaScript polyfils the role/states are allowed.

**Comment by lemnis** [comment]
I then would propose rephrasing it,  so it is more inline to other comments, e.g.

---

 Role: button (when used in conjunction with aria-expanded).

**Note:** The `aria-expanded` should reflect the existence of the `open` attribute within the parent `details` element

--- 

Still I think it is the reasoning is wrong, this is the only element that allows to set an implicit role as explicit as result of lack support. For example the `dialog` element is also not implemented by all browsers but does not allow an explicit role of `dialog`.

**Comment by carmacleod** [comment]
Perhaps the note could reuse some of the wording from a similar note for [`ol`](https://w3c.github.io/html-aria/#el-ol),  [`ul`](https://w3c.github.io/html-aria/#el-ul), etc. (please also see PR #206 for a suggested column change for those). So `summary` could have something like the following in the "implicit role" column:

> **Note:** some user agents don't support summary's implicit ARIA semantics. Authors can use role=button to set the role if necessary, with aria-expanded=true if the parent (details) element's open attribute is present, aria-expanded=false otherwise.

If the above note is moved to the "implicit role" column, then "**No** `role`" would need to be added to the "ARIA roles, states and properties which MAY be used" column for [`summary`](https://w3c.github.io/html-aria/#el-summary), to specify that no other roles are allowed (since the implicit role of button is already covered).

Does [details](https://w3c.github.io/html-aria/#el-details) need a similar note? Or do all user agents give it role=group by default?

Of course, it may be worth considering removing the note altogether 10 days from now (Jan 15), because Chromium Edge will support details/summary, and IE is... well, IE.

**Comment by scottaohara** [comment]
with chromium edge out in the wild now, it seems that we should reconsider this.

yes, it'll still be needed for IE11.  but if authors are polyfilling IE11, they should expect some conformance warnings (e.g. no role needed for landmark elements).

**Comment by LJWatson** [comment]
I agree it's worth reconsidering - not least because the present advice 
isn't really clear. It says the implicit button role shouldn't be used, 
but then includes button with aria-expanded in the permitted roles.
